src/config.rs

```diff
@@ -12,12 +13,24 @@ pub(crate) struct Config {
 }
 
 impl Config {
-    pub fn load() -> Result<Self, Error> {
-        let Ok(file) = File::open("config.json") else {
+    pub fn load(default_config_path: Option<PathBuf>) -> Result<Self, Error> {
+        let override_config_path = PathBuf::from("config.json");
+
+        let final_path = if override_config_path.is_file() {
```

Comment:
Use [`Path::exists`](https://doc.rust-lang.org/std/path/struct.Path.html#method.exists) here, so this can be a symlink too.

Reply:
[`is_file()`](https://doc.rust-lang.org/std/path/struct.PathBuf.html#method.is_file) already traverses symlinks

Reply:
Ok